client/src/components/Post/Post.tsx

```diff
@@ -266,11 +294,27 @@ class Post extends Component<Props, State> {
                     </div>
                     <div className={'body-footer'}>
                         <div className={'like-container'}>
-                            <MdThumbUp size={20} color={'#00B090'} className={'like-icon'}/>
+                            { this.state.clickedUpvote&&
+                                <MdThumbUp size={20} color={'#00B090'} className={'like-icon'}
+                                           onClick={() => this.handleUpvote()}/>
```

Comment:
```suggestion
                                           onClick={ this.handleUpvote}/>
```

Do not create arrow functions on the event handlers of an element inside a render... this will be creating a new function on every render! degrading performance